app/src/main/java/com/example/task1/ui/dashboard/DashboardActivity.kt

```diff
@@ -7,74 +6,115 @@ import android.util.Log
 import android.widget.ImageView
 import android.widget.TextView
 import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
 import androidx.recyclerview.widget.GridLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import androidx.viewpager2.widget.ViewPager2
 import com.example.task1.*
+import com.example.task1.data.models.Results
+
 import com.example.task1.ui.adapters.BannerAdapter
-import com.example.task1.ui.adapters.ProductAdapter
+import com.example.task1.ui.adapters.ItemsAdapter
 import com.google.android.material.tabs.TabLayout
 import com.google.android.material.tabs.TabLayoutMediator
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
 
-class DashboardActivity : AppCompatActivity(), ProductAdapter.OnCartClickListener {
+class DashboardActivity : AppCompatActivity(), ItemsAdapter.OnCartClickListener {
     private lateinit var viewPager: ViewPager2
-    private lateinit var dashboardViewModel: DashBoardViewModel
+    private lateinit var dashboardViewModel: DashboardViewModel
     private lateinit var tabLayout: TabLayout
-    private lateinit var recyclerViewForProduct: RecyclerView
+    private lateinit var recyclerviewForProduct: RecyclerView
     private lateinit var cartIcon: ImageView
     private lateinit var cartQuantityTextView: TextView
-    private var totalQuantityInCart: Int = 0
+   private var totalQuantityInCart: Int = 0
+
 
 
     @SuppressLint("MissingInflatedId")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_dash_board_screen)
+
         initView()
         fetchBanners()
```

Comment:
fetchBanners and fetchItems shouldn't be a separate thing anymore since they are one api. Instead fetchAllApi data at once and then use that data separately for banners and items. This goes for view model and repository as well, api is being called twice there.

Reply:
Yes it is done.